chicken-sloths/bangazon-cli-sprint2
# Tiny bug squashes

# Description
This is a smorgasbord of a PR! It fixes several tiny bug tickets, enumerated below.

## Related Ticket(s)
Fixes #56 (exits without setting an active customer)
Fixes #38 (prevents users from activating a nonexistent customer)
Fixes #60 (prevents users from selecting a nonexistent product type when they create a new product)

## Problem to Solve
Tiny bugs!

## Proposed Changes
- Added to the conditional on line 39 in `ui.js` to include option 12 (exit Bangazon) in the options you can do without selecting an active customer
- Refactored `views/activeCustV.js` so that it accepts the total number of customers in the database and returns a prompt object
- Added a `conform` method to `views/activeCustV.js` that checks to see if the customer id is greater than 0 and less than the total number of customers in the database. 
- Fixed regex in `views/addCustProdV.js` so that it only accepts digits 0-9. (If we ever added more product types, we'd need to refactor... but I figure that's a 'cross that bridge when we come to it situation, yes)

## Steps to Test Solution and Expected Results

First of all, run that sweet test suite. Everything should pass. Then...

Test #56 
1.  Run `npm start`
1. Don't select an active customer!! Resist the urge!!
1. Select option 12 to exit the program
**Expected result: you should be able to exit the program**

Text #38 
1. Run `npm start`
1. Select option 2
1. Try selecting a bunch of customer id's that you don't see listed. Examples: `-1`, `55`, `100000000000000000`, `Fred`
**Expected result: it should throw an error and not let you select those customers!**
1. Now practice selecting a customer id that does exist. Try it with one digit and two digit numbers. 
**Expected result: you should be able to select any customer id from that list**

Test #60 
1. Once you've selected a valid active customer id, select option 4 (add a product to sell)
1. Try typing in product type id's that aren't on the list (`-1`, `10`, `10000000`)


## Testing

- [ x] I certify that all existing tests pass

## Documentation
